internal/http.go

```diff
@@ -0,0 +1,239 @@
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/operate-first/curator-operator/pkg/signals"
+	"github.com/spf13/viper"
+
+	"github.com/jackc/pgx/v4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// Server defines the http server configuration
+type Server struct {
+	Addr        string
+	ErrorLog    *log.Logger
+	BaseContext func(net.Listener) context.Context
+}
+
+// HTTPError defines the http error codes
+type HTTPError struct {
+	code    int
+	message string
+}
+
+func InitializeRouter(ctx context.Context) {
```

Comment:
The value of this function seems like it's constructing a custom type. Can we rename this function to be something like `NewHTTPServer` or `NewAPIServer` and that function is responsible for building up a custom HTTP client type that contains any of the requisite configuration, and returns a built-up custom type that any call sites can use?